### PR TITLE
Replace uses of `strings.Split(N)` with `strings.Cut()`

### PR DESCRIPTION
**Pull request by thaJeztah**
`strings.Cut()` was added in go1.18, so now that go1.17 is EOL, it should be ok to use. For some of these I was initially a bit on the fence, but `strings.Cut` also is more performant, so switching made sense to do;

```go
func BenchmarkSplit(b *testing.B) {
	b.ReportAllocs()
	data := []string{"12hello=world", "12hello=", "12=hello", "12hello"}
	for i := 0; i < b.N; i++ {
		for _, s := range data {
			_ = strings.SplitN(s, "=", 2)[0]
		}
	}
}

func BenchmarkCut(b *testing.B) {
	b.ReportAllocs()
	data := []string{"12hello=world", "12hello=", "12=hello", "12hello"}
	for i := 0; i < b.N; i++ {
		for _, s := range data {
			_, _, _ = strings.Cut(s, "=")
		}
	}
}
```

    BenchmarkSplit
    BenchmarkSplit-10    	 8244206	       128.0 ns/op	     128 B/op	       4 allocs/op
    BenchmarkCut
    BenchmarkCut-10      	54411998	        21.80 ns/op	       0 B/op	       0 allocs/op

I split it up per package, but see individual commits for details

### CODE REVIEW

**Comment by thaJeztah** [comment]
Oh, interesting; didn't know it also tested the output for examples;

```
=== FAIL: github.com/docker/docker/api/types/filters ExampleArgs_MatchKVList (0.00s)
=== RUN   ExampleArgs_MatchKVList
true
false
true
false
got:

want:
true
false
true
false
```

**Review by cpuguy83 — APPROVED** [review]

File: api/types/types.go [hunk 1]
```diff
@@ -348,20 +348,19 @@ func DecodeSecurityOptions(opts []string) ([]SecurityOpt, error) {
...
 				return nil, fmt.Errorf("invalid security option %q", s)
 			}
-			if kv[0] == "" || kv[1] == "" {
+			if k == "" || v == "" {
 				return nil, errors.New("invalid empty security option")
 			}
-			if kv[0] == "name" {
-				secopt.Name = kv[1]
+			if k == "name" {
+				secopt.Name = k
```
[the remark this answers is not in the document]
Comment (thaJeztah) [reply api/types/types.go]: whoops; fixed!

File: daemon/graphdriver/zfs/zfs_freebsd.go [hunk 2]
```diff
@@ -24,15 +24,16 @@ func checkRootdirFs(rootdir string) error {
...
+const maxlen = 12
 
+func getMountpoint(id string) string {
 	// we need to preserve filesystem suffix
-	suffix := strings.SplitN(id, "-", 2)
-
-	if len(suffix) > 1 {
-		return id[:maxlen] + "-" + suffix[1]
+	id, suffix, _ := strings.Cut(id, "-")
+	if len(id) > 12 {
```
[the remark this answers is not in the document]
Comment (thaJeztah) [reply daemon/graphdriver/zfs/zfs_freebsd.go]: Well, it panics if `id` would be shorter than 12; https://go.dev/play/p/IHd0vxxaYh2

<details>

```go
// You can edit this code!
// Click here and start typing.
package main

import (
	"fmt"
	"strings"
)

func main() {
	fmt.Println(getMountpoint("123456789012-"))
	fmt.Println(getMountpoint("12345678901-"))
}

const maxlen = 12

func getMountpoint(id string) string {
	id, suffix, _ := strings.Cut(id, "-")
	id = id[:maxlen]
	if suffix != "" {
		// preserve filesystem suffix.
		id += "-" + suffix
	}
	return id
}
```

</details>

File: daemon/graphdriver/zfs/zfs_freebsd.go [hunk 3]
```diff
@@ -24,15 +24,16 @@ func checkRootdirFs(rootdir string) error {
...
+		id = id[:maxlen]
 	}
```
[the remark this answers is not in the document]
Comment (thaJeztah) [reply daemon/graphdriver/zfs/zfs_freebsd.go]: See above 😅

**Review by thaJeztah — COMMENTED** [review]
@tianon updated; PTAL

**Review by tianon — APPROVED** [review]

File: cmd/dockerd/daemon.go [hunk 4]
```diff
@@ -662,13 +662,11 @@ func loadListeners(cli *DaemonCli, serverConfig *apiserver.Config) ([]string, er
 
 	for i := 0; i < len(serverConfig.Hosts); i++ {
 		protoAddr := serverConfig.Hosts[i]
-		protoAddrParts := strings.SplitN(serverConfig.Hosts[i], "://", 2)
-		if len(protoAddrParts) != 2 {
+		proto, addr, ok := strings.Cut(protoAddr, "://")
+		if !ok || addr == "" {
 			return nil, fmt.Errorf("bad format %s, expected PROTO://ADDR", protoAddr)
```
Comment (thaJeztah) [inline cmd/dockerd/daemon.go:667]: Turned out this one was overly restrictive, as (contrary to the error message) `ADDR` is optional here (for `fd://` scheme); see  https://github.com/moby/moby/pull/44823

**Pull request merged**